src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -785,6 +785,19 @@ SDCPUThread::populateDependencies(shared_ptr<InflightInst> inst_ptr)
 
     // END Sequential Consistency
 
+    // BEGIN Only store if instruction is at head of inflightInsts
+
+    if (inflightInsts.size() > 1 && static_inst->isStore()) {
+        const shared_ptr<InflightInst> last_inst = *(++inflightInsts.rbegin());
+
+        if (!(last_inst->isCommitted() || last_inst->isSquashed()))
```

Comment:
Yea that's another weird thing that should really be fixed for all these cases, including the serializing instructions. IIRC, there were bugs with dangling dependencies before which were the reason the isCommitted || isSquashed check is all over the populateDependencies function...

Reply:
The proper logic should really be "if the previous instruction is squashed, look at the instruction before that one," and tbh most squashed instructions shouldn't even remain on the buffer, some just have to right now to hold onto the fault it generated... The case shouldn't be hit, it's just something that hasn't been changed yet... since earlier on I figured there might be reason to squash an instruction in the middle of the buffer without removing it...